File: nova-gestion-backend/src/main/java/nova/gestion/model/Task.java
package nova.gestion.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.sql.Timestamp;

@Data
@AllArgsConstructor
public class Task {
    private int idTask;
    private int idWorkSheet;
    private TaskType taskType;
    private Timestamp startTime;
    private Timestamp endTime;

    @JsonCreator
    public Task(int idTask, int idWorkSheet, Timestamp startTime, Timestamp endTime)
    {
        this.idTask = idTask;
        this.startTime = startTime;
        this.endTime = endTime;
        this.idWorkSheet = idWorkSheet;
    }
}

File: webui/src/app/models/task.ts
import {TaskType} from './task-type';
import {User} from './user.model';

export class Task {
  idTask: number;
  startTime: string;
  endTime: string;
  taskType: TaskType;
  idWorkSheet: number;
  duration: string;
  user: User;
}
